.circleci/config.yml

```diff
@@ -1,37 +1,319 @@
+  # -------------------------------------------------------------------------
+  hw_build:
+    <<: *job_defaults
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
 
-      - run: yarn install --frozen-lockfile --non-interactive
+      - run: yarn gulp ci:hw
 
-      - save_cache:
+  # -------------------------------------------------------------------------
+  # Job that makes sure Angular Layout can be integrated with the demo app
+  # -------------------------------------------------------------------------
+  aot_build:
+    <<: *job_defaults
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: yarn gulp ci:aot
+
+  # ----------------------------------
+  # Lint job. Runs the gulp lint task.
+  # ----------------------------------
+  lint:
+    <<: *job_defaults
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: yarn gulp ci:lint
+
+  # -------------------------------------------------------------------------------------------
+  # Job that builds all release packages with Gulp. The built packages can be then used in the
+  # same workflow to publish snapshot builds or test the dev-app with the release packages.
+  # -------------------------------------------------------------------------------------------
+  build_release_packages:
+    <<: *job_defaults
+    resource_class: xlarge
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: yarn gulp ci:build-release-packages
+
+      # Store the release output in the workspace storage. This means that other jobs
+      # in the same workflow can attach the release output to their job.
+      - persist_to_workspace:
+          root: dist/releases
           paths:
-            - node_modules
-          key: v1-dependencies-{{ checksum "yarn.lock" }}
+            - "**/*"
+
+      # Since there is no UMD bundle that includes everything, we need to move
+      # all bundles into a directory. This allows us to store all UMD bundles as job
+      # artifacts that can be picked up by the Angular Github bot.
+      - run:
+          name: Prepare artifacts for publish.
+          command: |
+            mkdir -p /tmp/layout-umd-minified-bundles
+            cp dist/releases/flex-layout/bundles/*.umd.min.js /tmp/layout-umd-minified-bundles
+      # Publish bundle artifacts which will be used to calculate the size change.
+      # Note: Make sure that the size plugin from the Angular robot fetches the artifacts
+      # from this CircleCI job (see .github/angular-robot.yml). Additionally any artifacts need to
+      # be stored with the following path format: "{projectName}/{context}/{fileName}"
+      # This format is necessary because otherwise the bot is not able to pick up the
+      # artifacts from CircleCI. See:
+      # https://github.com/angular/github-robot/blob/master/functions/src/plugins/size.ts#L392-L394
```

Comment:
Are you actually using the Github robot?

This seems to be out of scope for this PR because there is no bot configuration yet.

Reply:
I think we'd like to start, if anything to track size regressions. I figure we could add the config here and then onboard with the robot later?

Reply:
Yeah that works.